mage/mage.go

```diff
@@ -0,0 +1,280 @@
+/*
+Copyright 2021 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package mage
+
+import (
+	"fmt"
+	"log"
+	"net/url"
+	"os"
+	"path"
+	"path/filepath"
+	"strings"
+
+	"github.com/blang/semver"
+	"github.com/carolynvs/magex/pkg"
+	"github.com/carolynvs/magex/shx"
+	"github.com/pkg/errors"
+
+	kpath "k8s.io/utils/path"
+	"sigs.k8s.io/release-utils/command"
+)
+
+const (
+	// golangci-lint
+	defaultGolangCILintVersion = "v1.38.0"
+	golangciCmd                = "golangci-lint"
+	golangciConfig             = ".golangci.yml"
+	golangciURLBase            = "https://raw.githubusercontent.com/golangci/golangci-lint"
+
+	// zeitgeist
+	defaultZeitgeistVersion = "v0.2.0"
+	zeitgeistCmd            = "zeitgeist"
+	zeitgeistModule         = "sigs.k8s.io/zeitgeist"
+)
+
+// Ensure golangci-lint is installed and on the PATH.
+func EnsureGolangCILint(version string, forceInstall bool) error {
+	found, err := pkg.IsCommandAvailable(golangciCmd, version)
+	if err != nil {
+		return errors.Wrap(
+			err,
+			fmt.Sprintf("checking if %s is available", golangciCmd),
+		)
+	}
+
+	if !found || forceInstall {
```

Comment:
Let me know if you discover more patterns for installing dev tools that you think apply to multiple projects. 

I think what I have in the pkg is too simplistic for some cases, such as: force install, or allowing a range of acceptable versions on the client, or extracting a binary from a tgz on github releases.

Maybe if EnsurePackage took an Options struct with various behaviors, we could avoid having people reimplement these common scenarios in each project.

p.s. I'm super excited that you are switching to mage! 💃

Reply:
@carolynvs -- Thanks so much for creating the extensions!

I think an `Options` struct would be awesome!

Common use cases I've seen thus far:
- I need a tarball
- I need to install a go binary via go get (and it should go in `GOBIN`)
- I need to install a go binary (and I might want it in tmp or repo-specific path)
- I need to install something from GitHub (and it might not follow a neat filename convention)